torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -75,14 +75,10 @@ def step(self):
         Performs a single optimization step (parameter update).
         """
         self.optim.step()
-        for param_group in self.param_groups:
-            for params in param_group["params"]:
-                if params.grad is None:
-                    continue
-                self.averager.average_parameters(iter(params))
+        self.averager.average_parameters(params=self.param_groups)
```

Comment:
Actually I also found this bug recently, and here is the fix: https://github.com/pytorch/pytorch/pull/74737

The problem is `average_parameters` should have been only called exactly once per optimizer step, and otherwise the built-in step counter inside the averager will over-increase.

Reply:
yes, the counter has problem. but i fix it because this loop harms training speed when warm up steps. so i  move this loop into average_parameters()   and the iter a torch.nn.parameter.Paramete will return a new copy as mentioned before with one unit test

Reply:
>  this loop harms training speed when warm up steps

Your are right! No need to do unnecessary flattening here during the warmup stage. Agree with you now. We should let `average_parameters` also accept parameter groups for simplicity.

Please update the docstring of this method: "Average parameters or parameter groups of an optimizer". Thanks!

Reply:
fix already, please cr again

Reply:
Looks good. Thanks!